src/core/a-animation.js

```diff
@@ -373,6 +379,19 @@ function getAnimationValues (el, attribute, dataFrom, dataTo, currentValue) {
   };
 
   /**
+   * Match the scema type to color
+   * @return {bool} if the schema is of type color
+   */
+  function isColor () {
+    var componentName = attributeSplit[0];
+    var propertyName = attributeSplit[1];
+    var component = el.components[componentName];
+    var schema = component && component.schema;
+    if (schema) { return schema[propertyName].type === 'color'; }
```

Comment:
@JohnRodney Yes, I think this PR is very ad-hoc for the color type. The animation code is one of the most confusing areas in aframe and any ideas to improve it are more than welcome. In your approach, Would you pass the variable you want interpolated and the end value? Do you infer the type from the end value?